tests/test_extractors.py

```diff
@@ -0,0 +1,99 @@
+import attrs
+import pytest
+from parsel import Selector
+from web_poet import field
+
+from zyte_common_items import (
+    ProductFromList,
+    ProductFromListExtractor,
+    ProductFromListSelectorExtractor,
+    ProductVariant,
+    ProductVariantExtractor,
+    ProductVariantSelectorExtractor,
+)
+
+
+@pytest.mark.asyncio
+async def test_product_from_list_extractor():
+    @attrs.define
+    class MyProductFromListExtractor(ProductFromListExtractor):
+        data: dict
+
+        @field
+        def price(self):
+            return self.data["price"]
+
+        @field
+        def regularPrice(self):
+            return self.data["regularPrice"]
+
+    data = {"price": Selector("10€"), "regularPrice": Selector("20€")}
+    extracted = await MyProductFromListExtractor(data).to_item()
+
+    assert isinstance(extracted, ProductFromList)
+    assert extracted.price == "10.00"
+    assert extracted.regularPrice == "20.00"
+
+
+@pytest.mark.asyncio
+async def test_product_from_list_selector_extractor():
+    class MyProductFromListSelectorExtractor(ProductFromListSelectorExtractor):
+        @field
+        def price(self):
+            return self.css("price")
+
+        @field
+        def regularPrice(self):
+            return self.css("regularPrice")
+
+    selector = Selector(
+        "<data><price>10€</price><regularPrice>20€</regularPrice></data>"
+    )
+    extracted = await MyProductFromListSelectorExtractor(selector).to_item()
+
+    assert isinstance(extracted, ProductFromList)
+    assert extracted.price == "10.00"
+    assert extracted.regularPrice == "20.00"
+
+
+@pytest.mark.asyncio
+async def test_product_variant_extractor():
+    @attrs.define
+    class MyProductVariantExtractor(ProductVariantExtractor):
+        data: dict
+
+        @field
+        def price(self):
+            return self.data["price"]
+
+        @field
+        def regularPrice(self):
+            return self.data["regularPrice"]
+
+    data = {"price": Selector("10€"), "regularPrice": Selector("20€")}
```

Comment:
Let's have another case where another key exists in the dat input that is not declared as a `@field`.

Reply:
Could you elaborate on the aim of that? Would https://github.com/zytedata/zyte-common-items/pull/69/commits/3f8d115227f6e7a44cfca13a25f2ccf55988a867 work?

Reply:
I was thinking along the lines of the data being `{"price": "10€", "regularPrice": "20€", "other-data": "value"}` while the `@fields` are only `price` and `regularPrice`.

Reply:
https://github.com/zytedata/zyte-common-items/pull/69/commits/53a3c7378a84bd0d0a29aa03d16bfafc3fd4a3ec